toolkits/github/arcade_github/tools/activity.py

```diff
@@ -38,3 +38,56 @@ async def set_starred(
 
     action = "starred" if starred else "unstarred"
     return f"Successfully {action} the repository {owner}/{name}"
+
+
+# Implements https://docs.github.com/en/rest/activity/starring?apiVersion=2022-11-28#list-stargazers
+# Example `arcade chat` usage: "list the stargazers for the ArcadeAI/arcade-ai repo"
+@tool(requires_auth=GitHub())
+async def list_stargazers(
+    context: ToolContext,
+    owner: Annotated[str, "The owner of the repository"],
+    repo: Annotated[str, "The name of the repository"],
+    limit: Annotated[
+        Optional[int],
+        "The maximum number of stargazers to return. If not provided, all stargazers will be returned.",
+    ] = None,
+) -> Annotated[dict, "A dictionary containing the stargazers for the specified repository"]:
+    """List the stargazers for a GitHub repository."""
+    url = get_url("repo_stargazers", owner=owner, repo=repo)
+    headers = get_github_json_headers(context.authorization.token)
+
+    per_page = 30
```

Comment:
Can this be bigger? (more efficient)
If so, then I'd set `per_page` to be e.g. `min(limit, 100)`

Reply:
Thanks, addressed